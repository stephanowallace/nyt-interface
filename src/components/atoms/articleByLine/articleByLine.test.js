import React from 'react';
import { render, shallow } from 'enzyme';
import ArticleByLine from '.';

describe('Article ByLine', () => {
  it('renders loader', () => {
    const component = render(<ArticleByLine loading>test</ArticleByLine>);
    expect(component).toMatchSnapshot();
    expect(component.find('.loader')).toHaveLength(1);
  });

  it('renders content', () => {
    const component = render(<ArticleByLine>test</ArticleByLine>);
    expect(component.find('span').text()).toBe('test');
    expect(component).toMatchSnapshot();
  });

  it('accepts className prop', () => {
    const component = shallow(<ArticleByLine className="testClass">test</ArticleByLine>);
    expect(component.find('.testClass').exists()).toBeTruthy();
  });

  it('rejects not defined props', () => {
    const component = shallow(<ArticleByLine test="test">test</ArticleByLine>);
    expect(component.props().test).toBeUndefined();
  });
});
