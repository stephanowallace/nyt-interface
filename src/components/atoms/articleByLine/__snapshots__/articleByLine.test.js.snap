// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article ByLine renders content 1`] = `
<div
  class="byLine "
>
  <span
    class="container "
  >
    test
  </span>
</div>
`;

exports[`Article ByLine renders loader 1`] = `
<div
  class="byLine "
>
  <div
    class="container "
    style="width:100px;height:11px"
  >
    <div
      class="loader"
    />
  </div>
</div>
`;
